### ISSUE TITLE
[FE] 10-10EZ - SIP: Implement SIP functionality in Dependents section L&L pattern

### ISSUE THREAD

**Issue by hdjustice** [department-of-veterans-affairs/va.gov-team#95284]
## Benefit to the Veteran
As an authenticated Veteran, I want all my information that I input on the 10-10EZ saved if I need to leave the form, so that I can return later to complete and submit it, without having to re-enter some of the info that did not save.

## Background
Since the end of October 2023, we started to see submission failures for the 10-10EZ, failing all retries. The error message states Missing required fields and then lists the fields missing: a combination of Child DOB, SSN and Dependent date.
There has also been several failures with error message undefined method [] for nil:NilClass, which seems to indicate the Dependent name is missing.

We have since removed the functionality of "Save in progress" within the Dependents and Insurance sections. The Veteran must complete adding all dependents and/or insurance policies within a single session before it will save.

Now, we need to determine how to implement a validation mechanism on fields within the Dependents and Insurance sections, after a Veteran has saved the form (or auto-save), left the form and is now returning to the form.

## Scope
- **In-Scope**
     - Implement the Save-in-Progress functionality within the Household Dependents section
     - Display alerts on the Dependent cards if there are "*Required" fields that are missing input/blank
     - Allow the fields to be updated
     - SIP functionality automatically updates as it does on other pages
- **Out of Scope**
     - Any pre-fill opportunities for other Household section fields
     - Changes to fields or questions
     - Changes to question flow

## Resources
- [Figma Design](https://www.figma.com/design/UljiHam46o5DItC5iDgmPd/10-10EZ?node-id=0-13663&node-type=canvas&t=CPd4xDpz3QwiOE4C-0)
- [Design exploration ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/70217)
- [Mural Board](https://app.mural.co/t/departmentofveteransaffairs9999/m/departmentofveteransaffairs9999/1699485613760/d40578dbce4f0433d792988db24e4dec8d5b4fd1?sender=u5ad49c107baa41137f271007)

---
## Acceptance Criteria/Definition of Done
- The Save-in-progress functionality will be available within the Dependents List & Loop pattern

**Comment by hdjustice** [comment]
Still pending L&L work from other team to resolve the issues that are blocking this work (see Insurance tickets for additional details)

**Comment by rhasselle-oddball** [comment]
I believe this should be unblocked now. The [conditional logic being resolved](https://github.com/department-of-veterans-affairs/VA.gov-team-forms/issues/2026) should fix all major issues with pages not rendering properly presenting the correct options and validation. This will also make sure required fields are properly presented and validated at the time the user encounters them.

**Comment by hdjustice** [comment]
Thanks @rhasselle-oddball !  I will share with the team!

**Comment by hdjustice** [comment]
Ticket below was opened for future DST work to return a Veteran to their last accessed page within the loop.  This is not launch blocking for our work, we will proceed with our changes.
- [[arraybuilder] save in progress does not return to where you left off in middle of loop](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4490)

**Comment by allanto-ah** [comment]
Environment: RI

@longmd When I try to save my progress a second time, the "Finish this later" link is not showing. Ia that expected?

Test steps:
1. Go to dependents section and click to add a dependent
2. Fill out first and last name
3. Scroll down and click on "Finish this later" 
4. Click to Continue the form
5. Scroll down and notice that there isn't a "Finish this later" link

https://github.com/user-attachments/assets/56e629af-9fe2-4ce2-aed9-b636952c1f11

**Comment by longmd** [comment]
Okay... this feels like another forms bug ticket. I'm additionally curious if we see this only inside the arraybuilder, or if you tried it on another random form page if the behavior is the same?

**Comment by allanto-ah** [comment]
Environment: RI

Verified that SIP is working for dependents. 

There is one issue that Matt might take a look into.

The SIP is not triggering in a certain scenario, but this may be more of a platform bug
https://github.com/user-attachments/assets/6ff888bc-845a-410f-ae91-b94cffc6925c
